logstash-core/lib/logstash/util/safe_uri.rb

```diff
@@ -37,20 +38,117 @@ def inspect
   end
 
   def sanitized
-    return uri unless uri.password # nothing to sanitize here!
+    return uri unless password # nothing to sanitize here!
     
-    safe = uri.clone
-    safe.password = PASS_PLACEHOLDER
-    safe
+    user_info = user ? "#{user}:#{PASS_PLACEHOLDER}" : nil
+
+    # This syntax is required for jruby 1.7.x which doesn't reflect here correctly
+    # 9k does reflect here correctly. This can be removed when we are fully jruby 9k 
+    constructor = java.net.URI.java_class.constructor(java.lang.String, java.lang.String, java.lang.String, Java::int, java.lang.String, java.lang.String, java.lang.String)
+    constructor.new_instance(scheme, user_info, host, port, path, query, fragment)
   end
 
   def ==(other)
     other.is_a?(::LogStash::Util::SafeURI) ? @uri == other.uri : false
   end
 
   def clone
-    cloned_uri = uri.clone
-    self.class.new(cloned_uri)
+    # No need to clone the URI, in java its immutable
+    self.class.new(uri)
+  end
+
+  def update(field, value)
+    parts = {
```

Comment:
not sure it's worth creating this entire hash just to use its individual items later:
```ruby
@uri = java.net.URI.new(parts[:scheme], user_info, parts[:host], parts[:port], parts[:path], parts[:query], parts[:fragment])
```